python/feathub/feathub_client.py

```diff
@@ -28,6 +28,57 @@
 from feathub.table.table_descriptor import TableDescriptor
 
 
+class ProcessorJobGroup:
+    """
+    Group multiple feature materialization and execute them as one processor job.
+    """
+
+    def __init__(self, processor: Processor):
+        self._processor = processor
+        self._materialize_job_descriptors: List[MaterializeJobDescriptor] = []
+
+    def materialize_features(
+        self,
+        feature_descriptor: Union[str, TableDescriptor],
+        sink: FeatureTable,
+        ttl: Optional[timedelta] = None,
+        start_datetime: Optional[datetime] = None,
+        end_datetime: Optional[datetime] = None,
+        allow_overwrite: bool = False,
+    ) -> None:
+        """
+        :param feature_descriptor: Describes the table of features to be inserted in the
+                                   sink. If it is a string, it refers to the name of a
+                                   table descriptor in the entity registry.
+        :param sink: Describes the location to write the features.
+        :param ttl: Optional. If it is not None, the features data should be purged from
+                    the sink after the specified period of time.
+        :param start_datetime: Optional. If it is not None, the `features` table should
+                               have a timestamp field. And only writes into sink those
+                               features whose timestamp >= floor(start_datetime).
+        :param end_datetime: Optional. If it is not None, the `features` table should
+                             have a timestamp field. And only writes into sink those
+                             features whose timestamp <= ceil(start_datetime).
+        :param allow_overwrite: If it is false, throw error if the features collide with
+                                existing data in the given sink.
+        """
+        self._materialize_job_descriptors.append(
+            MaterializeJobDescriptor(
+                feature_descriptor=feature_descriptor,
+                sink=sink,
+                ttl=ttl,
+                start_datetime=start_datetime,
+                end_datetime=end_datetime,
+                allow_overwrite=allow_overwrite,
+            )
+        )
+
+    def start(self) -> ProcessorJob:
```

Comment:
In reference to the behavior of Flink's `StatementSet#execute`, it might be better to clear the materialized jobs after this method is invoked.